rct_optimizations/test/pnp_utest.cpp

```diff
@@ -0,0 +1,43 @@
+#include <rct_optimizations/experimental/pnp.h>>
+#include <gtest/gtest.h>
+
+using namespace rct_optimizations;
+
+TEST(PnPTest, 3DValidation)
+{
+  //correspondences: do easy offset between target & camera (ie make set translate 0.5 +x)
+  //setup params
+  PnPProblem3D setup;
+
+  //testing with only one observation
+  Correspondence3D3D::Set corrs;
+  corrs.reserve(9);
+
+  Eigen::Vector3d origin(0.0,0.0,0.0);
+
+  for(std::size_t i=0; i < 3; ++i)
+  {
+    for (std::size_t j =0; j < 3; j++)
+    {
+      Correspondence3D3D c;
+      Eigen::Vector3d world_point =Eigen::Vector3d(double(i), double(j), double((i*j)%3));
+      Eigen::Vector3d camera_point = world_point;
+      c.in_target = world_point;
+      c.in_image = camera_point;
+
+      corrs.push_back(c);
+    }
+  }
+
+  setup.correspondences = corrs;
+  setup.camera_to_target_guess = Eigen::Isometry3d::Identity();
+
+  PnPResult res = rct_optimizations::optimize(setup);
+  EXPECT_TRUE(res.converged);
```

Comment:
We should also expect:
1. The initial error to be extremely low (1.0e-15), since everything about the setup was perfect
1. The final error to be extremely low (1.0e-15), since everything about the setup was already perfect
1. The resulting transform should be almost exactly the same as the input (i.e. `EXPECT_TRUE(res.camera_to_target.isApprox(setup.camera_to_target)`)